core/libp2p/protocol_muxer/multiselect/connection_state.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/protocol_muxer/multiselect/connection_state.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::protocol_muxer,
+                            ConnectionState::ReadWriteError, e) {
+  using E = libp2p::protocol_muxer::ConnectionState::ReadWriteError;
+  switch (e) {
+    case E::PARTIAL_WRITE:
+      return "less bytes were written than expected";
+  }
+  return "unknown error";
+}
+
+namespace libp2p::protocol_muxer {
+  ConnectionState::ConnectionState(
```

Comment:
This looks scary. Might it be better to create a small builder for it?

Reply:
This ctor is just for `std::make_shared`, it isn's directly used anywhere, so it can be left like that.